app/assets/javascripts/pretty-text/engines/discourse-markdown/watched-words-replace.js

```diff
@@ -10,9 +10,17 @@ function findAllMatches(text, matchers, useRegExp) {
   const matches = [];
 
   if (useRegExp) {
+    const maxMatches = 100;
+    let index = 0;
```

Comment:
I would not call this the `index` but rather `count` since it's counting the number of matches?